File: Zestawy Zadań/C++/5 - Funkcje, rekurencja/Zestaw 2 zadania/5.19 Funkcja o prototypie void NORM(double X[], int n).cpp
#include <iostream>

double S5(int n) {
    if (n == 1) {
        return 1;
    } else {
        return 1.0 / (n * (n + 1)) + S5(n - 1);
    }
}

int main() {
    int N;
    std::cout << "Podaj liczbê ca³kowit¹: ";
    std::cin >> N;
    std::cout << "Wynik: " << S5(N) << std::endl;
    return 0;
}

File: Zestawy Zadań/Python/5 - Funkcje, rekurencja/Zestaw 1 zadania/5.19 Funkcja o prototypie void NORM(double X[], int n).py
def S5(n):
    if n == 1:
        return 1
    else:
        return 1 / (n * (n + 1)) + S5(n - 1)

if __name__ == "__main__":
    N = int(input("Podaj liczbę całkowitą: "))
    print("Wynik:", S5(N))
